torch/distributions/kl.py

```diff
@@ -258,13 +258,21 @@ def _kl_pareto_pareto(p, q):
     return result
 
 
+@register_kl(Poisson, Poisson)
+def _kl_poisson_poisson(p, q):
+    return p.rate * (p.rate.log() - q.rate.log()) - (p.rate - q.rate)
```

Comment:
We can make this one log using (p.rate / q.rate).log()

Reply:
Hmm, I usually intentionally take log before subtracting because it's numerically more stable (i.e. `p.rate/`q.rate` could overflow, while the subtraction does not). I'm fine with changing it, but I don't think the performance difference will be noticeable.